arch/sim/src/sim/up_linuxi2cbus.c

```diff
@@ -0,0 +1,332 @@
+/****************************************************************************
+ * arch/sim/src/sim/up_linuxi2cbus.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/types.h>
+#include <sys/ioctl.h>
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <syslog.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "up_linuxi2cbus.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define ERROR(fmt, ...) \
+        syslog(LOG_ERR, "up_linuxi2c_host: " fmt "\n", ##__VA_ARGS__)
+#define INFO(fmt, ...) \
+        syslog(LOG_ERR, "up_linuxi2c_host: " fmt "\n", ##__VA_ARGS__)
+#define DEBUG(fmt, ...)
+
+/* Define the Linux IOCTL interfaces again here so we do not require having
+ * the kernel headers to build this.  This interface should be fairly
+ * stable so it should not cause any issues.  The naming is preserved in case
+ * we want to change it in the future.
+ */
+
+/* Linux msg flags */
```

Comment:
it's better to include linux/i2c-dev.h to avoid the duplicated definition

Reply:
I really don't want to pull that header in here, both because of the additional build requirements and side effects of the header.  These are quite stable.

Reply:
Ok, but as you said this interface is very stable, why not include it directly:)? Since i2c interface isn't too complex and stable, the manual redeclaration is workable, but how about XWindow or ALSA which have many API and complex struct?

Reply:
We can try, I have just had issues in the past with kernel headers causing side effects.  I have gone back and forth, but since it is an advanced feature the extra requirement is not that bad long as it still works. .

Reply:
including them now.